vllm/entrypoints/openai/api_server.py

```diff
@@ -488,6 +498,48 @@ async def create_score_v1(request: ScoreRequest, raw_request: Request):
     return await create_score(request, raw_request)
 
 
+TASK_HANDLERS = {
+    "generate": {
+        "messages": (ChatCompletionRequest, create_chat_completion),
+        "default": (CompletionRequest, create_completion),
+    },
+    # should `embed` be a pooling task by default?
+    "embed": {
+        "messages": (EmbeddingChatRequest, create_embedding),
+        "default": (EmbeddingCompletionRequest, create_embedding),
+    },
+    "score": {
+        "messages": (PoolingChatRequest, create_score),
+        "default": (PoolingCompletionRequest, create_score),
```

Comment:
It should be `ScoreRequest` here.

Reply:
Done, have also removed the `messages` variant since scoring models require pure text pairs.